scheduler/src/cook/mesos/api.clj

```diff
@@ -1154,13 +1161,17 @@
                      (let [requested-guuids (->> (get-in ctx [:request :query-params "uuid"])
                                                  vectorize
                                                  (mapv #(UUID/fromString %)))
-                           not-found-guuids (remove #(group-exists? (db conn) %) requested-guuids)]
-                       (if (empty? not-found-guuids)
-                         [false {::guuids requested-guuids}]
+                           allow-partial-results (get-in ctx [:request :query-params :partial])
+                           exists? #(group-exists? (db conn) %)
+                           existing-groups (filter exists? requested-guuids)]
+                       (if (or
+                             (= (count existing-groups) (count requested-guuids))
+                             (and allow-partial-results (pos? (count existing-groups))))
+                         [false {::guuids existing-groups}]
```

Comment:
I moved the code to a new function, `retrieve-groups`, and added a docstring there